build/source/engine/time_utils.f90

```diff
@@ -174,5 +175,92 @@ subroutine compjulday(iyyy,mm,id,ih,imin,dsec,&  ! input
 
  end subroutine compjulday
 
+ ! ***************************************************************************************
+ ! public subroutine compgregcal: convert julian day (units of days) to calendar date
+ ! source: https://en.wikipedia.org/wiki/Julian_day#Julian_or_Gregorian_calendar_from_Julian_day_number
+ ! ***************************************************************************************
+
+ subroutine compcalday(julday,                              & !input
+                       iyyy,mm,id,ih,imin,dsec,err,message)   !output
+ USE multiconst,only:secprmin  ! seconds in an (day, hour, minute)
+ implicit none
+
+ ! input variables	
+ real(dp), intent(in)          :: julday       ! julian day
+
+ ! output varibles
+ integer(i4b), intent(out)     :: iyyy         ! year
+ integer(i4b), intent(out)     :: mm           ! month
+ integer(i4b), intent(out)     :: id           ! day
+ integer(i4b), intent(out)     :: ih           ! hour
+ integer(i4b), intent(out)     :: imin         ! minute
+ real(dp),     intent(out)     :: dsec         ! seconds
+ integer(i4b), intent(out)     :: err          ! error code
+ character(*), intent(out)     :: message      ! error message
+
+ ! local parameters
+ integer(i4b),parameter       :: y = 4716
+ integer(i4b),parameter       :: j = 1401
+ integer(i4b),parameter       :: m = 2
+ integer(i4b),parameter       :: n = 12
+ integer(i4b),parameter       :: r = 4
+ integer(i4b),parameter       :: p = 1461
+ integer(i4b),parameter       :: v = 3
+ integer(i4b),parameter       :: u = 5
+ integer(i4b),parameter       :: s = 153
+ integer(i4b),parameter       :: w = 2
+ integer(i4b),parameter       :: b = 274277
+ integer(i4b),parameter       :: c = -38
+ real(dp),parameter           :: hr_per_day = 24.0_dp
```

Comment:
I now recognize that these may be part of different calendar calculations. I would probably leave them (redefining as parameters is not really going to help as most are used only once). Provide the source of these calculations if possible. In VIC we have a similar routine, but also provide the references. If the values are only used once and only used here, it may even be clearer to just leave the literals in the equations. See for example https://github.com/UW-Hydro/VIC/blob/develop/vic/drivers/shared_all/src/vic_time.c